Failed: Http2ServerSpec should finally close after deadline (while closing request stream) against Akka 2.6.x

E.g. https://github.com/akka/akka-http/runs/5401287745?check_suite_focus=true

```
 [info]   - should finally close after deadline (while closing request stream) *** FAILED *** (69 milliseconds)
[info]     java.lang.AssertionError: assertion failed: expected class akka.stream.testkit.TestSubscriber$OnError, found class akka.stream.testkit.TestSubscriber$OnComplete$ (OnComplete)
[info]     at scala.Predef$.assert(Predef.scala:279)
[info]     at akka.testkit.TestKitBase.expectMsgClass_internal(TestKit.scala:572)
[info]     at akka.testkit.TestKitBase.expectMsgType(TestKit.scala:543)
[info]     at akka.testkit.TestKitBase.expectMsgType$(TestKit.scala:542)
[info]     at akka.testkit.TestKit.expectMsgType(TestKit.scala:973)
[info]     at akka.stream.testkit.TestSubscriber$ManualProbe.expectError(StreamTestKit.scala:486)
[info]     at akka.http.impl.engine.ws.ByteStringSinkProbe$$anon$1.expectError(ByteStringSinkProbe.scala:102)
[info]     at akka.http.impl.engine.http2.Http2ServerSpec$$anon$88.<init>(Http2ServerSpec.scala:1631)
```

--- a/akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2Demux.scala
+++ b/akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2Demux.scala
@@ -499,6 +499,7 @@ private[http2] abstract class Http2Demux(http2Settings: Http2CommonSettings, ini
         case CompletionTimeout =>
           info("Timeout: Peer didn't finish in-flight requests. Closing pending HTTP/2 streams. Increase this timeout via the 'completion-timeout' setting.")
 
+          shutdownStreamHandling()
           completeStage()
       }
 

